components/brave_wallet/browser/sns_resolver_task.cc

```diff
@@ -0,0 +1,665 @@
+  task_error_ = std::move(task_error);
+}
+
+void SnsResolverTask::ScheduleWorkOnTask() {
+  base::SequencedTaskRunnerHandle::Get()->PostTask(
+      FROM_HERE, base::BindOnce(&SnsResolverTask::WorkOnTask,
+                                weak_ptr_factory_.GetWeakPtr()));
+}
+
+void SnsResolverTask::WorkOnTask() {
+  if (!GetWorkOnTaskForTesting().is_null()) {
+    GetWorkOnTaskForTesting().Run(this);
+  }
+
+  if (task_result_) {
+    std::move(done_callback_).Run(this, std::move(task_result_), absl::nullopt);
+    // `this` is not valid here
+    return;
+  }
+  if (task_error_) {
+    std::move(done_callback_).Run(this, absl::nullopt, std::move(task_error_));
+    // `this` is not valid here.
+    return;
+  }
+
+  if (!domain_address_) {
+    domain_address_ = GetDomainKey(domain_, false);
+    if (!domain_address_) {
+      SetError(MakeInvalidParamsError());
+      ScheduleWorkOnTask();
+      return;
+    }
+  }
+
+  // Check if there is an nft token associated with domain.
+  if (!nft_owner_check_done_) {
+    if (!nft_mint_address_) {
+      nft_mint_address_ = GetMintAddress(*domain_address_);
+      if (!nft_mint_address_) {
+        NftOwnerDone(absl::nullopt);
+        ScheduleWorkOnTask();
+        return;
+      }
+    }
+
+    // Check if domain is minted and its supply is 1.
+    if (!nft_mint_supply_check_done_) {
+      FetchNftSplMint();
+      return;
+    }
+
+    // Find token account for that mint and extract owner from its data.
+    FetchNftTokenOwner();
+    return;
+  }
+
+  // No nft. Find owner of domain and contents of domain's SOL record.
+  if (!domain_name_registry_state_) {
+    FetchDomainRegistryState();
+    return;
+  }
+
+  if (!sol_record_name_registry_state_) {
+    FetchSolRecordRegistryState();
+    return;
+  }
+}
+
+void SnsResolverTask::SetError(SnsResolverTaskError error) {
+  task_error_.emplace(std::move(error));
+}
+
+void SnsResolverTask::NftOwnerDone(absl::optional<SolanaAddress> nft_owner) {
+  nft_owner_check_done_ = true;
+  if (nft_owner) {
+    task_result_.emplace(*nft_owner);
+  }
+}
+
+void SnsResolverTask::FetchNftSplMint() {
+  DCHECK(domain_address_);
+  DCHECK(!nft_owner_check_done_);
+  DCHECK(!nft_mint_supply_check_done_);
+
+  auto internal_callback = base::BindOnce(&SnsResolverTask::OnFetchNftSplMint,
+                                          weak_ptr_factory_.GetWeakPtr());
+  RequestInternal(solana::getAccountInfo(nft_mint_address_->ToBase58()),
+                  std::move(internal_callback),
+                  solana::ConverterForGetAccountInfo());
+}
+
+void SnsResolverTask::OnFetchNftSplMint(APIRequestResult api_request_result) {
+  ScopedWorkOnSnsTask work_on_task(this);
+
+  if (!api_request_result.Is2XXResponseCode()) {
+    SetError(MakeInternalError());
+    return;
+  }
+
+  absl::optional<SolanaAccountInfo> account_info;
+  if (!solana::ParseGetAccountInfo(api_request_result.body(), &account_info)) {
+    SetError(MakeInternalError());
```

Comment:
Should it be `SetError(ParseErrorResult(api_request_result.body()));` instead?

Reply:
https://github.com/brave/brave-core/pull/15617/commits/1e34bd9363c2e1aa4593c00f3409b8c119ccab29